cmd/kubeadm/app/preflight/checks.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+)
+
+type PreFlightError struct {
+	Msg   string
+	Count int
+}
+
+func (e *PreFlightError) Error() string {
+	return fmt.Sprintf("PreFlight Error\n  count: %d \n  msg: %s", e.Count, e.Msg)
```

Comment:
nit: Pre-flight error or just pre-flight error

(in some parts of the k8s code it's best practice to have lowercase error messages, but we don't have to obey that here I think)